Jellyfin.Plugin.Bookshelf/Providers/ComicVine/BaseComicVineProvider.cs

```diff
@@ -0,0 +1,226 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Extensions.Json;
+using Jellyfin.Plugin.Bookshelf.Providers.ComicVine.Models;
+using MediaBrowser.Common.Net;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Plugin.Bookshelf.Providers.ComicVine
+{
+    /// <summary>
+    /// Base class for the Comic Vine providers.
+    /// </summary>
+    public abstract class BaseComicVineProvider
+    {
+        private const string IssueIdMatchGroup = "issueId";
+
+        private readonly ILogger<BaseComicVineProvider> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IComicVineMetadataCacheManager _comicVineMetadataCacheManager;
+        private readonly IComicVineApiKeyProvider _apiKeyProvider;
+
+        private static readonly Regex[] _issueIdMatches = new[]
+        {
+            // The slug needs to be stored in the provider id for the IExternalId implementation
+            new Regex(@"^(?<slug>.+?)\/(?<issueId>\d+-\d+)$"),
```

Comment:
it's recommended to use `[0-9]` instead of `\d` since `\d` includes numbers in all locales (recommend fixing all occurrences)